src/lib.rs

```diff
@@ -99,6 +100,112 @@ where
     namespace: BTreeMap<String, AmlValue>,
 }
 
+/// The pointer to the EBDA (Extended Bios Data Area) start segment pointer
+const EBDA_START_SEGMENT_PTR: usize = 0x40e;
+/// The earliest (lowest) memory address an EBDA (Extended Bios Data Area) can start
+const EBDA_EARLIEST_START: usize = 0x80000;
+/// The end of the EBDA (Extended Bios Data Area)
+const EBDA_END: usize = 0x9ffff;
+/// The start of the main bios area below 1mb in which to search for the RSDP
+/// (Root System Description Pointer)
+const RSDP_BIOS_AREA_START: usize = 0xe0000;
+/// The end of the main bios area below 1mb in which to search for the RSDP
+/// (Root System Description Pointer)
+const RSDP_BIOS_AREA_END: usize = 0xfffff;
+/// The RSDP (Root System Description Pointer)'s signature, "RSD PTR " (note trailing space)
+const RSDP_SIGNATURE: &'static [u8; 8] = b"RSD PTR ";
+
+/// Find the begining of the EBDA (Extended Bios Data Area) and return `None` if the ptr at
+/// `0x40e` is invalid.
+fn find_ebda_start<H>(handler: &mut H) -> Option<usize>
+where
+    H: AcpiHandler,
+{
+    // Read base segment from BIOS area. This is not always given by the bios, so it needs to be
+    // checked. We left shift 4 because it is a segment ptr.
+    let base_mapping = handler.map_physical_region::<u16>(
+        EBDA_START_SEGMENT_PTR, mem::size_of::<u16>()
+    );
+    let base = (*base_mapping as usize) << 4;
+    handler.unmap_physical_region(base_mapping);
+
+    // Check if base segment ptr is in valid range valid
+    if (EBDA_EARLIEST_START..EBDA_END).contains(&base) {
+        debug!("EBDA address is {:#x}", base);
+        Some(base)
+    } else {
+        warn!(
+            "EBDA address at {:#x} out of range ({:#x}), falling back to {:#x}",
+            EBDA_START_SEGMENT_PTR,
+            base,
+            EBDA_EARLIEST_START
+        );
+
+        None
```

Comment:
Could we collect the logic here and later on and refactor this into a `find_ebda` that just returns the range of addresses. Instead of returning `None` when we can’t find the pointer, just return the `EBDA_EARLIEST_START..=EBDA_END` that we fall back to anyways.

Reply:
Yeah, makes more sense than how I have it now